src/oaklib/implementations/ontoportal/ontoportal_implementation_base.py

```diff
@@ -50,22 +49,18 @@ def __post_init__(self):
         if self.focus_ontology is None:
             if self.resource:
                 self.focus_ontology = self.resource.slug
-        if not self.OntoPortalClientClass:
-            raise NotImplementedError("OntoPortalClientClass not specified")
-        api_key = get_apikey_value(self.api_key_name)
-        self.client = self.OntoPortalClientClass(api_key=api_key)
+        if not self.ontoportal_client_class:
+            raise NotImplementedError("ontoportal_client_class not specified")
+        api_key = get_apikey_value(self.ontoportal_client_class.name)
```

Comment:
the API key getting is handled by the class itself, so you can skip this completely from the OAK side

Reply:
Yeah I saw that ontoportal_client manages API keys as well but as I understand it that would require the user to create either an environment variable or a file under `~/.config`. The OAK command line interface has a `set-apikey` subcommand that I think we need to continue to support. So I thought it was better to continue to let OAK manage its own key storage.

Reply:
In https://github.com/INCATools/ontology-access-kit/pull/142 I made it much more integrated with PyStow, since it's a pain to have to use several competing systems when there's already something ready to go out of the box. If someone wants to put effort into unmucking the testing (#173), then it will be possible to finish that PR, since right now the logging is too much of a mess to even figure it out. If I had to guess though, it might be because PRs from forks don't get access to the secrets in the GHA workflows

Reply:
I can talk to Sujay about #173 since he currently is assigned and see if has any progress on it. If not maybe I can take it over from him. Also after I commented on #142 Chris gave me admin rights on this repo, so I'd be happy to work with you on that one. But as far as these changes go are you okay with decoupling them from the PyStow work? I'm not sure how much effort that will be, but does it make sense to just get this in now?

Reply:
All of the code is simple, the only real effort is coordination :p

Reply:
I can clean this up later, it’s not a blocker for your PR

Reply:
Definitely appreciate all your feedback. I'm going to merge this now, but please get in touch with me next week (obo-community slack maybe?) and we can try to coordinate on getting #142 going.